src/Microsoft.Health.Fhir.Azure/IntegrationDataStore/AzureAccessTokenClientInitializerV2.cs

```diff
@@ -54,22 +75,21 @@ public async Task<CloudBlobClient> GetAuthorizedClientAsync(IntegrationDataStore
                 throw new IntegrationDataStoreClientInitializerException(Resources.InvalidStorageUri, HttpStatusCode.BadRequest);
             }
 
-            string accessToken;
             try
             {
-                accessToken = await _accessTokenProvider.GetAccessTokenForResourceAsync(storageAccountUri, cancellationToken);
+                return Task.FromResult(new BlobServiceClient(storageAccountUri, CreateDefaultTokenCredential()));
             }
             catch (AccessTokenProviderException atp)
             {
                 _logger.LogError(atp, "Unable to get access token");
 
                 throw new IntegrationDataStoreClientInitializerException(Resources.CannotGetAccessToken, HttpStatusCode.Unauthorized);
             }
+        }
 
-#pragma warning disable CA2000 // Dispose objects before losing scope
-            StorageCredentials storageCredentials = new StorageCredentials(new TokenCredential(accessToken));
-#pragma warning restore CA2000 // Dispose objects before losing scope
-            return new CloudBlobClient(storageAccountUri, storageCredentials);
+        private static TokenCredential CreateDefaultTokenCredential()
+        {
```

Comment:
The same credential as used in ExportClient's AzureAccessTokenClientInitializer. Please let me know if this is the right one. (I cannot test this with a local test run with the emulator.)